libraries/redcore/api/soap/helper.php

```diff
@@ -44,13 +44,227 @@ public static function createSoapFaultResponse($message, $faultCode = 'SOAP-ENV:
+						break;
+
+					default:
+						if (($resourceSpecific != '' && RApiHalHelper::attributeToString($resource, 'resourceSpecific') == $resourceSpecific)
+							|| $resourceSpecific == '')
+						{
+							if ($namesOnly)
+							{
+								$outputResources[] = RApiHalHelper::attributeToString($resource, 'displayName');
+							}
+							else
+							{
+								$resource->addAttribute('name', $resource['displayName']);
+								$outputResources[] = $resource;
+							}
+						}
+				}
+			}
+		}
+
+		return $outputResources;
+	}
+
+	/**
+	 * Returns the resoult resource from a certain operation
+	 *
+	 * @param   SimpleXMLElement  $xmlElement  Xml element
+	 *
+	 * @return  array
+	 */
+	public static function getResultResource($xmlElement)
+	{
+		if (isset($xmlElement->resources->resource))
+		{
+			foreach ($xmlElement->resources->resource as $resource)
+			{
+				$displayName = RApiHalHelper::attributeToString($resource, 'displayName');
+				$resourceSpecific = RApiHalHelper::attributeToString($resource, 'resourceSpecific');
+
+				if ($displayName == 'result' && $resourceSpecific == 'rcwsGlobal')
+				{
+					$resource->addAttribute('name', $resource['displayName']);
+
+					return $resource;
+				}
+			}
+		}
+
+		$resource = new SimpleXMLElement('<resource name="result" displayName="result" transform="array" fieldFormat="{result}" />');
+
+		return $resource;
+	}
+
+	/**
+	 * Method to determine the wsdl file name
+	 *
+	 * @param   string  $client          Client
+	 * @param   string  $webserviceName  Name of the webservice
+	 * @param   string  $version         Suffixes to the file name (ex. 1.0.0)
+	 * @param   string  $extension       Extension of the file to search
+	 * @param   string  $path            Path to webservice files
+	 *
+	 * @return  string  The full path to the api file
+	 *
+	 * @since   1.4
+	 */
+	public static function getWebserviceFile($client, $webserviceName, $version = '', $extension = 'xml', $path = '')
+	{
+		JLoader::import('joomla.filesystem.path');
+
+		if (!empty($webserviceName))
+		{
+			$version = !empty($version) ? JPath::clean($version) : '1.0.0';
+			$webservicePath = !empty($path) ? RApiHalHelper::getWebservicesRelativePath() . '/' . $path : RApiHalHelper::getWebservicesRelativePath();
+
+			$rawPath = $webserviceName . '.' . $version;
+			$rawPath = !empty($extension) ? $rawPath . '.' . $extension : $rawPath;
+			$rawPath = !empty($client) ? $client . '.' . $rawPath : $rawPath;
+
+			return $webservicePath . '/' . $rawPath;
+		}
+
+		return '';
+	}
+
+	/**
+	 * Select resources from output array to display them in SOAP output list
+	 *
+	 * @param   array  $outputResources  Selected output resources from the ws xml config file
+	 * @param   array  $items            Output resources with final values
+	 *
+	 * @return  array  Array of selected resources and value in simple array for SOAP output
+	 *
+	 * @since   1.4
+	 */
+	public static function selectListResources($outputResources, $items)
```

Comment:
It's because I needed a simple array just for doing in_array comparisons.  I'm not doing any transforms here, just listing the resources by filtering from an initial array which actually comes from the hal helper